### ISSUE TITLE
Cannot Consume Scroped Service - DBContext

### ISSUE THREAD

**Issue by DanJ210** [aspnet/Docs#7404]
My site is easy. After making all changes up to the changes in startup which I did, I'm getting an unhandled exception. Cannot consume scoped service from singleton. The singleton is my DbContext hooked up to my repository. This site is already live and has been live and running for a while. I'm just trying to migrate 2.0 to 2.1 and I'm getting this after following the docs to the startup changes. What do I do?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4fa0f562-bdf0-eeea-f076-0c3e05222f75
* Version Independent ID: 8269d917-3b32-1607-71da-c756d4d7ad32
* Content: [Migrate from ASP.NET Core 2.0 to 2.1](https://docs.microsoft.com/en-us/aspnet/core/migration/20_21?view=aspnetcore-2.1)
* Content Source: [aspnetcore/migration/20_21.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/migration/20_21.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @DanJ210. I'm sorry to hear you're having trouble. Someone might see this and be able to provide a tip, but we generally recommend Stack Overflow for faster support. We are **very interested** when docs fail to give devs the info they need, so please do post back if you find that the problem with your app migration could have been solved by something missing or not described well in the migration topic.

Having said all of that ... one thing interesting about your comment is that you say the singleton is your dB context and your repository is throwing. Is your repository registered as a scoped service? Why isn't your dB context registered scoped (i.e., are you not registering it with `AddDbContext`)?

**Comment by DanJ210** [comment]
So I had to tediously update all the NuGet packages that I had installed to 2.1.0, if I did 2.1.1 it would fail. There was about 17 and this was after removing the `Microsoft.AspNetCore.All` to replace with `Microsoft.AspNetCore.App`. I had 19 updates from 2.1.0 to 2.1.1 so I had to keep selecting all and updating to view the output details which told me which NuGet package was missing and I had to install about 70 some NuGet packages individually one by one doing that update all process until there were no more output errors. Finally then I was able to update all 90 some NuGet packages at once from 2.1.0 to 2.1.1. Problem after that was `Unhandle Exception: No Service Type for 'Microsoft.ApplicationInsights.AspNetCore.JavaScriptSnippets' has been registered...`

After searching I found this article on the developer community, [developer community](https://developercommunity.visualstudio.com/content/problem/50735/applicationinsights-javascriptsnippet-exception-wi.html) which has one answer suggesting to remove two lines from the _layout.cshtml page that's generated by Visual Studio. These two lines I don't have any idea what they do but they obviously have something to do with application insights. I haven't seen anything about this in the docs and I'm unsure if this will break my insights or not but I guess we will see.


Would you guys like me to post on stack overflow still? So far my site is working. I'm afraid to publish to Azure but nothing seems broken. I've seemed to have successfully migrated to 2.1 and NuGet packages to 2.1.1 but I did have to remove those two lines which I'll paste below..

`@inject Microsoft.ApplicationInsights.AspNetCore.JavaScriptSnippet JavaScriptSnippet`
`@Html.Raw(JavaScriptSnippet.FullScript)`

**Comment by DanJ210** [comment]
@guardrex Anytime I get a yellow flag I update. Right now I'm 15.7. I looked through the other linked GitHub issues. Maybe someone will look at this and understand something else. I was just migrating from 2.0 to 2.1 from a project I had running and working and didn't add or change anything except all of the NuGet packages which I had to upgrade one at a time because I couldn't all at once.

So idk what to do here. I removed the two lines ago mentioned above and I really don't know what that did to the application.

**Comment by guardrex** [comment]
AFAIK `Microsoft.NETCore.App` is managed implicitly by the SDK, so make sure you have the latest SDK, which is 2.1.302. https://www.microsoft.com/net/download You shouldn't need to reference the package in a framework-dependent app.

The latest for `Microsoft.AspNetCore.App` is 2.1.2; but even if it's set to an earlier patch version (2.1.0, 2.1.1), it should roll-forward automatically to 2.1.2 if you have the latest SDK installed.

You don't need a package reference for `Microsoft.EntityFrameworkCore.SqlServer` ... that package is inside the `Microsoft.AspNetCore.App` metapackage ... https://www.nuget.org/packages/Microsoft.AspNetCore.App/ and https://docs.microsoft.com/aspnet/core/fundamentals/metapackage-app

Here's a tip for working with package versions and a failing app: Locally cached packages can cause problems when you're trying to troubleshoot package versions. After you get a new combo of package versions in place, you should execute `nuget locals all --clear` (if you have nuget.exe installed) or `dotnet nuget locals --clear all` (if you want to use the .NET Core command) before restoring packages for the app. The command will clear out your locally cached packages and thus force package restore from the NuGet package source (nuget.org).

**Comment by DanJ210** [comment]
Thank you @guardrex I appreciate your extra help on this for me with the extra tip. I'm installing 2.1.301 now as that makes sense. I'll see if I can then update the packages or since it was a brand new template, try creating a brand new project.

I thought 2.1 was doing away with all the included packages within app and we do need each individual. Maybe I have that backwards and that's how 2.0 was. I'll need to read over it again. If I have success with what I'm doing here I'll let you know so you can at least know that you helped me in many ways.

Thanks.

**Comment by DanJ210** [comment]
Yes that's what I meant. And you're right. After installing the latest 2.1.302 I no longer see the RC1 and I'm able to scaffold a database, hook up the context in startup.cs and reference EntityFrameworkCore without have to add the package.

Thanks very much for helping understand this. Sorry we got a little off track here but maybe my SDK had something to do with the issue above?

**Comment by DanJ210** [comment]
@guardrex is an invaluable tool here.
